addons/mrp_subcontracting/models/mrp_production.py

```diff
@@ -16,6 +17,24 @@ class MrpProduction(models.Model):
     )
     subcontracting_has_been_recorded = fields.Boolean("Has been recorded?", copy=False)
 
+    incoming_picking = fields.Many2one(related='move_finished_ids.move_dest_ids.picking_id')
+
+    @api.depends('name')
+    def name_get(self):
+        return [
+            (record.id, "%s (%s)" % (record.incoming_picking.name, record.name)) if record.bom_id.type == 'subcontract'
+            else (record.id, record.name) for record in self
+        ]
+
+    @api.model
+    def _name_search(self, name='', args=None, operator='ilike', limit=100, name_get_uid=None):
+        args = args or []
+        if operator == 'ilike' and not (name or '').strip():
+            domain = []
+        else:
+            domain = ['|', ('name', operator, name), ('incoming_picking.name', operator, name)]
+        return self._search(expression.AND([domain, args]), limit=limit, access_rights_uid=name_get_uid)
+
```

Comment:
You can probably clean it since https://github.com/odoo/odoo/pull/86588

Reply:
Done!